app/src/components/market/advanced_filters/index.tsx

```diff
@@ -39,57 +41,60 @@ const Options = styled(Dropdown)`
   max-width: 100%;
 `
 
-export const AdvancedFilters = () => {
-  const currencyOptions: Array<DropdownItemProps> = [
-    {
-      content: <TokenItem icon={<EtherIcon />} text="Ether" />,
-      onClick: () => {
-        console.warn('Option Ether')
-      },
-    },
-    {
-      content: <TokenItem icon={<DaiIcon />} text="DAI" />,
-      onClick: () => {
-        console.warn('Option DAI')
-      },
-    },
-    {
-      content: <TokenItem icon={<BatIcon />} text="Basic Atentio Token" />,
-      onClick: () => {
-        console.warn('Option Basic Atention Token')
-      },
+interface Props {
+  onChangeCurrency: (currency: Maybe<string>) => void
+  onChangeArbitrator: (arbitrator: Maybe<string>) => void
+  onChangeTemplateId: (templateId: Maybe<string>) => void
+}
+
+const currencyIcons: { [index: string]: JSX.Element } = {
+  BAT: <BatIcon />,
+  DAI: <DaiIcon />,
+  ETH: <EtherIcon />,
+}
+
+export const AdvancedFilters = (props: Props) => {
+  const context = useConnectedWeb3Context()
+  const { networkId } = context
+
+  const arbitrators = getArbitratorsByNetwork(networkId)
+  const tokens = getTokensByNetwork(networkId)
+
+  const { onChangeArbitrator, onChangeCurrency, onChangeTemplateId } = props
+
+  const currencyOptions: Array<DropdownItemProps> = [{ address: null, symbol: 'All' }, ...tokens].map(
+    ({ address, symbol }, index) => {
+      const icon = currencyIcons[symbol]
+      return {
+        content: <TokenItem icon={icon} text={symbol} />,
+        onClick: () => onChangeCurrency(address),
+      }
     },
-  ]
+  )
 
   const questionTypeOptions: Array<DropdownItemProps> = [
+    {
+      content: 'All',
+      onClick: () => onChangeTemplateId(null),
+    },
     {
       content: 'Binary',
-      onClick: () => {
-        console.warn('Option Binary')
-      },
+      onClick: () => onChangeTemplateId('0'),
```

Comment:
Template ids are numbers (actually big numbers, but I don't think that will be necessary soon).